### ISSUE TITLE
sample_lsm percentage_inside issue

### ISSUE THREAD

**Issue by ManuelSpinola** [r-spatialecology/landscapemetrics#324]
I am using sample_lsm with irregular areas, similar to the example, "Calculating landscape metrics for irregular areas" in the landscapemetrics website. However, I got percentage_inside values much higher than 100, even of 465.  Why will be the reason?

I can share the files if need it.

metrica_presente = sample_lsm(s_presente_cat, uf_aclap,
                                                      level = "class",
                                                      metric = "ca")

s_presente_cat is a categorical raster with 5 classes

uf_aclap is an sf object with 10 polygons.

**Comment by Nowosad** [comment]
Yes @ManuelSpinola -- please share the files

**Comment by ManuelSpinola** [comment]
https://1drv.ms/f/s!AmkQxwG61-VIlOdKiUOS2CQNjjMZLg?e=GvJKJH

**Comment by Nowosad** [comment]
@ManuelSpinola -- look at the image below. It shows one of the rasters cells with parts of the first multipolygon. Your data has many small polygons (much smaller than the cell) and thus the large percentage values.

![image](https://github.com/r-spatialecology/landscapemetrics/assets/3457131/5b2cbc11-6402-415f-ac89-30f982e8fa1f)

Maybe in your case, you could start by simplifying the polygon layer (https://r.geocompx.org/geometry-operations#simplification) before doing calculations.

**Comment by ManuelSpinola** [comment]
Thank you very much.

**Comment by mhesselbarth** [comment]
@Nowosad Do you think we need a warning for this very extreme case that the polygon is smaller than even one cell? It is quite an extreme case which I didn't even consider so far.

**Comment by Nowosad** [comment]
I do not think that warning is needed -- how about improving the docs: https://github.com/r-spatialecology/landscapemetrics/pull/325/files ?
